packages/core-components/src/components/LogViewer/LogLine.tsx

```diff
@@ -159,19 +159,28 @@ export interface LogLineProps {
   classes: ReturnType<typeof useStyles>;
   searchText: string;
   highlightResultIndex?: number;
+  setRowHeight?: (index: number, size: number) => void;
 }
 
 export function LogLine({
   line,
   classes,
   searchText,
   highlightResultIndex,
+  setRowHeight,
 }: LogLineProps) {
+  const lineRef = useRef<HTMLSpanElement>(null);
   const chunks = useMemo(
     () => calculateHighlightedChunks(line, searchText),
     [line, searchText],
   );
 
+  useEffect(() => {
+    if (lineRef.current && setRowHeight) {
+      setRowHeight(line.lineNumber, lineRef.current.offsetHeight);
+    }
+  }, [line.lineNumber, setRowHeight]);
```

Comment:
does this need `lineRef.current` as a dep too? Or is that a non-issue, ie is the ref always set at the point where the effect runs